### ISSUE TITLE
Describe EventRepositoryDecorator and its function

### ISSUE THREAD

**Issue by aslakknutsen** [arquillian/continuous-enterprise-development#69]
- CDI Decorator that decorates all Repository<T> instances to add eventing
  - Created / Removed
- Used by Services to hook into the lifecycles of objects created

``` java
public class MailServiceObserver {
  @Inject
  private MailService service;

  public void sendMailOnUserCreation(@Observes @Created User newUser) { 
    service.sendMail(newUser);
  }
}
```

**Comment by ALRubinger** [comment]
You're proposing I describe this where?  In the EJB Services chapter where the mail service is?

**Comment by ALRubinger** [comment]
Also I don't see this class anywhere?  We should be adding it?

**Comment by aslakknutsen** [comment]
https://github.com/arquillian/continuous-enterprise-development/blob/master/code/application/domain/core/src/main/java/org/cedj/geekseek/domain/EventRepositoryDecorator.java

"How to hook in behavior into the domain lifecycle" or something.. 

EJB is not really relevant here.

Maybe it should be described close to where the Repository is described. With a reference like; 'we'll see why/how we need/use this when we implement our mail service in chap x' ?

**Comment by ALRubinger** [comment]
Thanks, must have messed up my find query when looking last night.

**Comment by ALRubinger** [comment]
No, actually it was MailServiceObserver I didn't find.  What exactly do you want from this issue? :)

**Comment by aslakknutsen** [comment]
> No, actually it was MailServiceObserver I didn't find.

MailServiceObserver does not exist. It's an example of how to integrate the MailService into the 'app' when we want 'on register' emails sent. 

>  What exactly do you want from this issue? :)

Well, actually. You're the one who wanted the issue. It came up after our meeting where I explained how the rest / domain/services work together.  :)

**Comment by ALRubinger** [comment]
Ah, OK.  So basically the issue is "integrate these in code and then
explain it".

**Comment by ALRubinger** [comment]
From Aslak:

# [NOTE]
1. Explain EventRepositoryDecorator, and where that should be done
- CDI Decorator that decorates all Repository<T> instances to add eventing
  *\* Created / Removed
- Used by Services to hook into the lifecycles of objects created
  *\* MailService.x(@Observes @Created User) { sendMail() }
  *\* Services can choose to Observe During or After transcation commit
  ====

**Comment by aslakknutsen** [comment]
?

**Comment by ALRubinger** [comment]
I took your notes from the REST chapter and put them in this issue so I could address elsewhere.

**Comment by aslakknutsen** [comment]
but.. but.. those notes == this issue.. 

(except the last point, which I seemed to have missed :)

**Comment by ALRubinger** [comment]
Oh.  Yes.  It's frikkin 4:30AM here, give me a break.
